pyproject.toml

```diff
@@ -128,6 +128,7 @@ testpaths = [
 xfail_strict = true
 
 [tool.ruff]
+extend-exclude = ["examples"]
```

Comment:
Do we have to do this? Running pre-commit twice results in 6 failures total. Cleaner code makes it easier for new users to use IMO.

![image](https://github.com/user-attachments/assets/6211bc3d-a2b1-4419-b32a-5e850c715e80)

Reply:
I wouldn't consider these to be best practices for notebooks. Annotations, e.g., make this much harder do read. And a notebook is all about being easy to grasp, not type safety.

Reply:
Fair enough. But we also miss out on formatting as this is currently done.

Reply:
Agreed, but I'd point to the same thing: this needs to be tourist-friendly, which is often not maintainer-friendly. For example, sometimes you'll want to have really long lines to not take up vertical space with stuff that doesn't matter to illustrate a point, or you'll do weird alignments. Better to leave the examples free-form IMO.

Reply:
Okay. Well I suggest we leave it as it is, or exclude `examples` specifically from the `ruff` pre-commit hook

Reply:
Last thing, could we change it to this? Smaller diff when adding others
```suggestion
extend-exclude = [
	"examples",
]
```